ximea_ros2_cam/src/ximea_ros2_cam.cpp

```diff
@@ -66,19 +85,20 @@ XimeaROSCam::~XimeaROSCam() {
 
             
         //Save Camera context
-        char* cam_context=NULL;
-	    #define SIZE_OF_CONTEXT_BUFFER (10*1024*1024) // 10MiB
-	    cam_context = (char*)malloc(SIZE_OF_CONTEXT_BUFFER);
-	    xi_stat= xiGetParamString(this->xi_h_, XI_PRM_API_CONTEXT_LIST, cam_context, SIZE_OF_CONTEXT_BUFFER);
-
-        std::ofstream file("/home/m2s2/ecal_meas/cam_context.bin", std::ios::out | std::ios::binary);
-        file.write(cam_context, SIZE_OF_CONTEXT_BUFFER);
-        file.close();
-
-        RCLCPP_INFO_STREAM(this->get_logger(), "Done Writing camera context file");
-        std::string s(cam_context);
-        RCLCPP_INFO_STREAM(this->get_logger(), "Cam Context: " << s);
-        
+        if (!this->cam_context_path_.empty()){
+            char* cam_context=NULL;
+            #define SIZE_OF_CONTEXT_BUFFER (10*1024*1024) // 10MiB
```

Comment:
I don't know, I only include the condition !this->cam_context_path_.empty() to avoid unnecessary savings.

Reply:
As explained, the code inside the IF statement was not constructed by me, it is already in the code, I only include the IF to avoid save unnecessary files every run section.